src/language/tsLanguageService.ts

```diff
@@ -0,0 +1,215 @@
+      );
+      return '';
+    }
+
+    let content = contents[name];
+
+    if (typeof content !== 'string') {
+      let libPath;
+
+      /* if (name === `lib.${TS_CONFIG_LIBRARY_NAME}.d.ts`) {
+        libPath = join(this._extensionPath, 'node_modules/typescript/lib', name);
+      } else */
+
+      if (name === `${MDB_CONFIG_LIBRARY_NAME}.d.ts`) {
+        libPath = join(this._extensionPath, 'node_modules/mongodb', name);
+      } else if (name === `${GLOBAL_CONFIG_LIBRARY_NAME}.d.ts`) {
+        libPath = join(this._extensionPath, 'src/types', name);
+      } else {
+        content = '';
+      }
+
+      if (libPath) {
+        try {
+          content = readFileSync(libPath, 'utf8');
+        } catch (e) {
+          this._connection.console.error(
+            `Unable to load library ${name} at ${libPath}`
+          );
+          content = '';
+        }
+      } else {
+        content = '';
+      }
+
+      contents[name] = content;
+    }
+
+    return content;
+  }
+
+  _getTypeScriptServiceHost() {
+    const compilerOptions = {
+      allowNonTsExtensions: true,
+      allowJs: true,
+      lib: [
+        `${MDB_CONFIG_LIBRARY_NAME}.d.ts`,
+        `${GLOBAL_CONFIG_LIBRARY_NAME}.d.ts`,
+      ], // , `lib.${TS_CONFIG_LIBRARY_NAME}.d.ts`],
+      target: ts.ScriptTarget.Latest,
+      moduleResolution: ts.ModuleResolutionKind.Classic,
+      experimentalDecorators: false,
+    };
+    let currentTextDocument = TextDocument.create('init', 'javascript', 1, '');
+
+    // Create the language service host to allow the LS to communicate with the host.
+    const host: ts.LanguageServiceHost = {
+      getCompilationSettings: () => compilerOptions,
+      getScriptFileNames: () => [currentTextDocument.uri],
+      getScriptKind: () => {
+        return ts.ScriptKind.JS;
+      },
+      getScriptVersion: (fileName: string) => {
+        if (fileName === currentTextDocument.uri) {
+          return String(currentTextDocument.version);
+        }
+        return '1';
+      },
+      getScriptSnapshot: (fileName: string) => {
+        let text = '';
+        if (fileName === currentTextDocument.uri) {
+          text = currentTextDocument.getText();
+        } else {
+          text = this._loadLibrary(fileName);
+        }
+        return {
+          getText: (start, end) => text.substring(start, end),
+          getLength: () => text.length,
+          getChangeRange: () => undefined,
+        };
+      },
+      getCurrentDirectory: () => '',
+      getDefaultLibFileName: () => GLOBAL_CONFIG_LIBRARY_NAME,
+      readFile: (path: string): string | undefined => {
+        if (path === currentTextDocument.uri) {
+          return currentTextDocument.getText();
+        }
+        return this._loadLibrary(path);
+      },
+      fileExists: (): boolean => false,
+      directoryExists: (): boolean => false,
+    };
+
+    // Create the language service files.
+    const jsLanguageService = ts.createLanguageService(host);
```

Comment:
[Question only:] Correct me if I’m wrong, but the general structure here is that `host` serves as our way to connect the TS logic to our VSCode documents, but other than that, the `LanguageService` instance is unaware of the fact that it’s running against a VSCode environment/as part of a VSCode extension. Is that right? Could we potentially use this, say, inside of mongosh to provide autocompletions there?